### PR TITLE
implement MessageReader for Paho MQTT

### PR DESCRIPTION
**Pull request by Alfusainey**
A first patch that implements the `MessageReader` for #335 

**TODO**: implement the MessageWriter for MqttV3 and MqttV5 clients. 

@slinkydeveloper any feedback is welcome, while i work on implementing the MessageWriter interface

Signed-off-by: Alfusainey Jallow <alf.jallow@gmail.com>

### CODE REVIEW

File: mqtt/paho/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          <groupId>org.eclipse.paho</groupId>
+          <artifactId>org.eclipse.paho.mqttv5.client</artifactId>
+          <version>${paho.version}</version>
+      </dependency>
+      <!-- Paho MQTTv3 client -->
+	  <dependency>
+         <groupId>org.eclipse.paho</groupId>
+         <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
+         <version>${paho.version}</version>
+      </dependency>
```
Comment (slinkydeveloper) [inline mqtt/paho/pom.xml:41]: Why you import both, more than just v5? I see you're using only v5

Comment (Alfusainey) [reply mqtt/paho/pom.xml:41]: See the `MqttMessageFactory#createReader` factory methods and the corresponding `MqttMessageFactoryTest` class

Comment (slinkydeveloper) [reply mqtt/paho/pom.xml:41]: I see what you mean. I wonder, can we somehow develop this package in a way that it doesn't hardly depend on v3/v5, so a user don't need to bring both in their classpath? E.g. we could make both dependencies as "optional" and then, depending on what you have on the classpath, the encoder is able to pick one of the two? Does this idea vaguely make sense?

Comment (Alfusainey) [reply mqtt/paho/pom.xml:41]: yes, this makes a lot of sense. I can try that and update the patch

**Comment by liangyuanpeng** [comment]
It's good work and i want to know any update?  

and some reference, in my project, we use `hivemq-mqtt-client` and it contains v3 and b5.

**Comment by pierDipi** [comment]
Thanks for this PR, @Alfusainey!

Are you still interested in this contribution?

**Comment by Alfusainey** [comment]
> Thanks for this PR, @Alfusainey!
> 
> Are you still interested in this contribution?

@pierDipi actually this PR fell off my radar since i changed roles after I filed it. But yes, i am interested to see this through. I now have it in my todo list for next week

**Comment by Alfusainey** [comment]
> It's good work and i want to know any update?
> 
> and some reference, in my project, we use `hivemq-mqtt-client` and it contains v3 and b5.

@liangyuanpeng There is one piece left in this PR which is to implement the `MessageWriter` interface. I will work on this next week and hopefully it will be merged soon.

**Comment by JemDay** [comment]
BI .. it appears as though I've independently duplicated this PR, at least in spirit.

@Alfusainey - Is this a PR you're actively working on..

**Comment by Alfusainey** [comment]
@JemDay i was on holidays and resumed yesterday :-) 

> I rather stupidly didn't look through all the existing PR's prior to starting to code this.

This is my fault I think, because this PR should have been out of the way a long time ago, sorry!

> @Alfusainey - Is this a PR you're actively working on..

As you can see the discussions here, I am not actively working on this but I would like to see this PR merged soon. Please give me until mid February to complete it, otherwise please go ahead and provide your own implementation (possibly reusing whats here if you wish)

**Comment by Alfusainey** [comment]
Closing this PR in favor of #513. We can continue the discussions there...

**Pull request closed without merging**